File: src/tests/test_parser/Simple.py
from JavaParserVisitor import ClassInfo, MethodInfo

def expectedAnswer():
    ans = dict()
    mainClassInfo = ClassInfo("Simple")
    mainClassInfo.members.add("x")
    for letter in ["f", "g", "h", "i", "j", "k", "l"]:
        method = MethodInfo(letter)
        method.memberReferences.add("x")
        mainClassInfo.methods[letter] = method
    
    ans["Simple"] = mainClassInfo
    return ans

File: src/tests/test_parser/Simple.java
public class Simple {
  public int f() {
    int y = x + 21;
    return 42;
  }
  public void g() {
    x++;
  }
  public void h() {
    int x = x + 1;
  }
  public int i() {
    int x = 42;
    return this.x;
  }
  public void j() {
    this.x++;
  }
  public void k() {
    --this.x;
  }
  public void l() {
    --x;
  }
  int x = f();
}
